src/data_watcher.cpp

```diff
@@ -323,6 +327,24 @@ std::optional<DataOperation>
     return std::nullopt;
 }
 
+std::optional<DataOperation>
+    DataWatcher::processMovedTo(const EventInfo& receivedEventInfo)
+{
+    // Case 1 : If a file inside a configured and watching directory is renamed.
+    // Case 2 : A file is moved to a configured and watching directory.
+    // Case 3 : If a configured and watching file itself is renamed.
+    fs::path absCopiedPath =
+        _watchDescriptors.at(std::get<WD>(receivedEventInfo)) /
+        std::get<BaseName>(receivedEventInfo);
+
+    if (absCopiedPath.string().starts_with(_dataPathToWatch.string()))
+    {
+        lg2::debug("Processing IN_MOVED_TO for {PATH}", "PATH", absCopiedPath);
```

Comment:
Please include cookie so that we could associated IN_MOVED_TO/FROM event while debugging.

We will see a bunch of debug traces during PEL archive.